bitcoin/src/pow.rs

```diff
@@ -1389,6 +1401,29 @@ mod tests {
         .is_err()); // invalid length
     }
 
+    const HEX_LITERAL: &str = "0x01003456";
+
+    #[test]
+    fn compact_target_from_hex_str_happy_path() {
+        let actual = CompactTarget::from_hex_str(HEX_LITERAL).unwrap();
+        let expected = CompactTarget(u32::from_str_radix(HEX_LITERAL.trim_start_matches("0x"), 16).unwrap());
```

Comment:
This is not what I meant. I meant to let rustc parse the integer by writing `CompactTarget(0x01003456)`

Reply:
Oh duh, lol